modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java

```diff
@@ -80,66 +78,35 @@
  * <p>Additionally limits the lock map size.
  */
 public class HeapLockManager extends AbstractEventProducer<LockEvent, LockEventParameters> implements LockManager {
-    /** Throttled logger. */
-    private static final IgniteLogger THROTTLED_LOG = Loggers.toThrottledLogger(
-            Loggers.forClass(HeapLockManager.class),
-            // TODO: IGNITE-24181 Get rid of Common thread pool.
-            ForkJoinPool.commonPool()
-    );
-
-    /**
-     * Table size.
-     */
+    /** Table size. */
     public static final int DEFAULT_SLOTS = 1_048_576;
 
     private static final String LOCK_MAP_SIZE_PROPERTY_NAME = "lockMapSize";
 
-    private static final String RAW_SLOTS_MAX_SIZE_PROPERTY_NAME = "rawSlotsMaxSize";
-
-    /**
-     * Striped lock concurrency.
-     */
+    /** Striped lock concurrency. */
     private static final int CONCURRENCY = Math.max(1, Runtime.getRuntime().availableProcessors() / 2);
 
-    /** Lock map size. */
-    private final int lockMapSize;
+    private final AtomicInteger lockTableSize = new AtomicInteger();
 
-    /** Raw slots size. */
-    private final int rawSlotsMaxSize;
+    /** An unused state to avoid concurrent allocation. */
+    private LockState removedLockState;
```

Comment:
Why it's not static ?

Reply:
Because it uses a deadlock prevention policy configuration, which is available only on start.